sqlmesh/core/snapshot/definition.py

```diff
@@ -691,15 +696,20 @@ def missing_intervals(
                 start,
                 end,
                 strict=False,
+                allow_partial=allow_partial,
             )
         )
 
         interval_unit = self.node.interval_unit
 
-        upper_bound_ts = to_timestamp(
-            self.node.cron_floor(execution_time) if not ignore_cron else execution_time
-        )
-        end_ts = min(end_ts, to_timestamp(interval_unit.cron_floor(upper_bound_ts)))
+        if allow_partial:
+            upper_bound_ts = to_timestamp(execution_time)
+            end_ts = min(end_ts, upper_bound_ts)
```

Comment:
why are we taking the upper bound min? why not always do end_ts?

Reply:
why does that matter? their source data could have future dates

Reply:
yeah, but if the future end results in complete interval we will happily record it